doc/source/guidelines/private_packaging.rst

```diff
@@ -0,0 +1,78 @@
+.. _private_dependencies:
+
+
+Hosting Private Dependencies
+============================
+There will be cases in which it is necessary to host and pull packages that are
+not ready to be hosted to the public `PyPI`_. For example, if a PyAnsys library
+requires auto-generated gRPC interface files from an as-of-yet private feature
+or service, this package should be hosted on a private PyPI repository.
+
+Ansys has a private repository at `PyAnsys PyPI`_, and access is controlled via
```

Comment:
@akaszynski what are other companies doing which might have similar scenarios?

Reply:
Desire:
There is a huge push for packages to be hosted in GitHub packages, and I think the drive is to do that:
https://github.community/t/pypi-compatible-github-package-registry/14615/103

Solutions I've seen in the wild:
A common solution has been to use public ADO for hosting both universal artifacts and Python packages. Another alternative within GitHub is to use a repository for that:
https://github.com/astariul/github-hosted-pypi

While this solution works, it's non-optimal as you have to git push and we would still have to track a PAT as pulling would require access to a private repository.

Other alternatives include
- Using the [GitLab PyPI repository](https://docs.gitlab.com/ee/user/packages/pypi_repository/) (we don't use GitLab).
- [Hosting Private PyPI Server on Kubernetes](https://python.plainenglish.io/private-pypi-server-on-kubernetes-7df169864972) (we don't want to maintain our own PyPI server.

In the end we chose ADO to:
- Avoid the security risk external pulling from internal, having to create a new internal registry.
- Avoid creating our own private PyPI registry thereby opening up a new attack surface.
- Reuse a tried and true Microsoft backed service.